packages/dd-trace/src/llmobs/telemetry.js

```diff
@@ -1,12 +1,48 @@
 'use strict'
 
+const {
+  SPAN_KIND,
+  MODEL_PROVIDER,
+  PARENT_ID_KEY,
+  SESSION_ID,
+  INTEGRATION
+} = require('./constants/tags')
+
 const telemetryMetrics = require('../telemetry/metrics')
+
+const LLMObsTagger = require('./tagger')
+
 const llmobsMetrics = telemetryMetrics.manager.namespace('mlobs')
 
 function incrementLLMObsSpanStartCount (tags, value = 1) {
   llmobsMetrics.count('span.start', tags).inc(value)
 }
 
+function incrementLLMObsSpanFinishedCount (span, value = 1) {
+  const mlObsTags = LLMObsTagger.tagMap.get(span)
+
+  const isRootSpan = mlObsTags[PARENT_ID_KEY] !== ROOT_PARENT_ID
+  const hasSessionId = mlObsTags[SESSION_ID] !== null
```

Comment:
```suggestion
  const hasSessionId = mlObsTags[SESSION_ID] != null
```
this will match for `undefined` and `null`, where as strict equal `!==` will only pass for `null`, but not for `undefined` (gotta love js)